bibliopixel/led.py

```diff
@@ -18,9 +18,9 @@ def __init__(self, driver):
         self._data = []
         self._driver = driver
 
-    def setData(self, data):
+    def setData(self, colors, pos):
         self._reading.wait()
-        self._data = data
+        self._data = (colors, pos)
```

Comment:
Well, it's no less threadsafe than anything that was there before.  :-D

I guess making a copy does protect you a little bit from the consequences
of your actions if you are editing a list while you're trying to send it to
a driver, but either way there's a race condition.

However, this "race condition" doesn't result in undefined behavior.  It
only means that changes might "go off" in the middle of rendering.  This
doesn't sound bad and might even be positive.

On Mon, Jun 20, 2016 at 6:02 PM, adammhaile <notifications@github.com>
wrote:

> In bibliopixel/led.py
> <https://github.com/ManiacalLabs/BiblioPixel/pull/53#discussion_r67775250>
> :
>
> >          self._reading.wait()
> > -        self._data = data
> > +        self._data = (colors, pos)
>
> Oh... because using multi driver, each driver may get a subset of the full
> list. That makes sense. I assume that's all threadsafe?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ManiacalLabs/BiblioPixel/pull/53/files/f1413a5dbdcca6ea516504330b5eeec68eb7d502#r67775250>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAPdshX1c68mkkaxhOoiKcFz11D-m-Zrks5qNw4DgaJpZM4I6Iga>
> .
>



-- 
     /t

http://radio.swirly.com - art music radio 24/7 366/1000